### PR TITLE
Jira issue 128: SND - Update SND Event related tables to require SNDViewerPermission

### PR DESCRIPTION
**Pull request by spamhurts**
#### Changes

- Change AdminPermission requirement to SNDViewerPermission on: EventsCacheTable, PackageAttributeTable, AttributeDataTable, and EventNotesTable
- Add SNDViewerPermission to EventsTable
- Update Javascript event related tests (Add SNDEditorRole to test user roles)

### CODE REVIEW

**Review by jallentxbiomed — APPROVED** [review]

File: src/org/labkey/snd/SNDUserSchema.java [hunk 1]
```diff
@@ -136,15 +136,19 @@ public TableInfo createTable(SNDUserSchema schema, ContainerFilter cf)
                     @Override
                     public TableInfo createTable(SNDUserSchema schema, ContainerFilter cf)
                     {
-                        return new EventsTable(schema, SNDSchema.getInstance().getTableInfoEvents(), cf).init();
+                        if (schema.getContainer().hasPermission(schema.getUser(), SNDViewerPermission.class, schema.getContextualRoles()))
+                        {
+                            return new EventsTable(schema, SNDSchema.getInstance().getTableInfoEvents(), cf).init();
+                        }
+                        return null;
```
Comment (RameshRapa) [inline src/org/labkey/snd/SNDUserSchema.java:143]: Can i please know how we are handleing this null

Comment (spamhurts) [reply src/org/labkey/snd/SNDUserSchema.java:143]: in the SNDManager.java starting at line 184:
```
   /**
     * Generic get table info function
     */
    public TableInfo getTableInfo(@NotNull UserSchema schema, @NotNull String table)
    {
        TableInfo tableInfo = schema.getTable(table);
        if (tableInfo == null)
            throw new IllegalStateException(table + " TableInfo not found");

        return tableInfo;
    }

```

Comment (RameshRapa) [reply src/org/labkey/snd/SNDUserSchema.java:143]: Thanks Terry but in this scenario we should be giving error message related to View/Admin permission. Is it still ok if we throw "TableInfo not found" for the user?

Comment (spamhurts) [reply src/org/labkey/snd/SNDUserSchema.java:143]: Yes. It is a generic function returning a generic message that is technically correct. The UI should prevent the user from trying to access the data if they don't have permission to do so. Jay is working on that code now in snprc_ehr. This permission check will keep them from getting to the data if they follow some other path such as navigating to the schema browser and selecting one of the tables.

**Review by RameshRapa — APPROVED** [review]

**Pull request merged**